app/backend/approaches/readretrieveread.py

```diff
@@ -142,8 +147,9 @@ def __init__(self, employee_name: str, callbacks: Callbacks = None):
                          name="Employee",
                          description="useful for answering questions about the employee, their benefits and other personal information",
                          callbacks=callbacks)
-        self.func = self.employee_info
+        self.func = lambda _: 'Not implemented'
+        self.coroutine = self.employee_info
         self.employee_name = employee_name
 
-    def employee_info(self, name: str) -> str:
+    async def employee_info(self, name: str) -> str:
```

Comment:
Yes, due to the agent being called with arun. Otherwise this is the error:

```
Traceback (most recent call last):
  File "/workspaces/azure-search-openai-demo/app/backend/app.py", line 90, in ask
    r = await impl.run(request_json["question"], request_json.get("overrides") or {})
  File "/workspaces/azure-search-openai-demo/app/backend/approaches/readretrieveread.py", line 134, in run
    result = await agent_exec.arun(q)
  File "/workspaces/azure-search-openai-demo/app/backend/backend_env/lib/python3.10/site-packages/langchain/chains/base.py", line 263, in arun
    return (await self.acall(args[0], callbacks=callbacks))[self.output_keys[0]]
  File "/workspaces/azure-search-openai-demo/app/backend/backend_env/lib/python3.10/site-packages/langchain/chains/base.py", line 178, in acall
    raise e
  File "/workspaces/azure-search-openai-demo/app/backend/backend_env/lib/python3.10/site-packages/langchain/chains/base.py", line 172, in acall
    await self._acall(inputs, run_manager=run_manager)
  File "/workspaces/azure-search-openai-demo/app/backend/backend_env/lib/python3.10/site-packages/langchain/agents/agent.py", line 1002, in _acall
    next_step_output = await self._atake_next_step(
  File "/workspaces/azure-search-openai-demo/app/backend/backend_env/lib/python3.10/site-packages/langchain/agents/agent.py", line 928, in _atake_next_step
    result = await asyncio.gather(
  File "/workspaces/azure-search-openai-demo/app/backend/backend_env/lib/python3.10/site-packages/langchain/agents/agent.py", line 909, in _aperform_agent_action
    observation = await tool.arun(
  File "/workspaces/azure-search-openai-demo/app/backend/backend_env/lib/python3.10/site-packages/langchain/tools/base.py", line 358, in arun
    raise e
  File "/workspaces/azure-search-openai-demo/app/backend/backend_env/lib/python3.10/site-packages/langchain/tools/base.py", line 330, in arun
    await self._arun(*tool_args, run_manager=run_manager, **tool_kwargs)
  File "/workspaces/azure-search-openai-demo/app/backend/backend_env/lib/python3.10/site-packages/langchain/tools/base.py", line 438, in _arun
    raise NotImplementedError("Tool does not support async")
NotImplementedError: Tool does not support async
```